tools/rum-models-generator/Sources/RUMModelsGeneratorCore/Input/Swift/JSONToSwiftTypeTransformer.swift

```diff
@@ -187,17 +189,27 @@ internal class JSONToSwiftTypeTransformer {
         return `struct`
     }
 
+    /// Returns `true` if the given `SwiftStruct.Property` contains a mutable property (`var`) or any of its nested types does.
+    private func transitiveMutability(of property: SwiftStruct.Property) -> SwiftStruct.Property.Mutability {
+        let innerMutability = transitiveMutableProperty(of: property.type)
+        if property.mutability.rawValue > innerMutability.rawValue {
+            return property.mutability
+        }
+        return innerMutability
+    }
+
     /// Returns `true` if the given `SwiftType` contains a mutable property (`var`) or any of its nested types does.
-    private func hasTransitiveMutableProperty(type: SwiftType) -> Bool {
+    private func transitiveMutableProperty(of type: SwiftType) -> SwiftStruct.Property.Mutability {
         switch type {
         case let array as SwiftArray:
-            return hasTransitiveMutableProperty(type: array.element)
+            return transitiveMutableProperty(of: array.element)
         case let `struct` as SwiftStruct:
-            return `struct`.properties.contains { property in
-                property.isMutable || hasTransitiveMutableProperty(type: property.type)
+            return `struct`.properties.reduce(.immutable) { max, property in
+                let innerMutability = transitiveMutability(of: property)
+                return innerMutability.rawValue > max.rawValue ? innerMutability : max
```

Comment:
I'm afraid this `.rawValue` comparison won't be so clear when visiting this code in the future - the problem we solve in `rum-models-generator` is quite abstract and examples and context are priceless 😉.

How about extracting it to a separate method with dedicated naming and more context given in its comment:
```swift
/// Returns new `Mutability` for parent type given its child `Mutability`.
///
/// This stands for: _if the child is mutable, its parent must be mutable too_;
/// e.g.: in `foo.bar.property = 2` expression, not only `property` must be mutable, but also its parent `bar` accessor.
private func resolve(parentMutability: Mutability, childMutability: Mutability) -> Mutability {
    return parentMutability.rawValue > childMutability.rawValue ? parentMutability : childMutability
}
```
This could be then reused from `transitiveMutableProperty(of type:)` and `transitiveMutability(of property:)`.

Reply:
That's a good idea! What do you thing of adding:
```swift
internal func max<T>(_ lhs: T, _ rhs: T) -> T where T: RawRepresentable, T.RawValue: Comparable {
    rhs.rawValue > lhs.rawValue ? rhs : lhs
}
``` 
to the `Utilities.swift`?

This way I could reuse that function for testing the `Mutability` level order.

Reply:
I think this wouldn't address the main concern on documenting what we do: generic `func max<T>()` is same vague as `>` operator, whereas plain `func` gives us the clear point for documenting the context in its comment.